### PR TITLE
3135 - 1

### PR DESCRIPTION
**Pull request by nikolaiprivalov**
## Номер
3135

## Номер задания
1

## Ссылка на видео с демо
https://www.youtube.com/watch?v=pjnOz-zZm0w

### CODE REVIEW

File: 3135/1/auxiliary.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+require 'YAML'
+require 'table_print'
+
+private
```
Comment (longbeardmen) [inline 3135/1/auxiliary.rb]: you should define private method in classes

File: 3135/1/auxiliary.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+def check_top(gems, options)
+  num = options[:top]
+  return gems[1..num] if num && gems.length > num
+
+  gems
+end
+
+def print_table(gems)
+  tp gems, :name, :used_by, :watch, :star, :fork, :issues, :contributors,
```
Comment (longbeardmen) [inline 3135/1/auxiliary.rb]: `tp` bad method name

File: 3135/1/auxiliary.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+private
+
+def load_list(options)
+  YAML.load_file(options[:file] || 'gem_list.yml')['gems']
+end
+
+def check_names(list, options)
+  seq = options[:name]
+  return list.select! { |name| name.include? seq } if seq
```
Comment (longbeardmen) [inline 3135/1/auxiliary.rb]: do you really need to change `list` object?

File: 3135/1/gem_class.rb [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  end
+
+  def github_link
+    @github = [@response['source_code_uri'], @response['homepage_uri']]
+              .find { |link| link.to_s.include? 'github.com' }
+  end
+
+  # rubocop:disable Metrics/AbcSize, Security/Open
+  # :reek:DuplicateMethodCall
+  def github_parse
```
Comment (longbeardmen) [inline 3135/1/gem_class.rb]: sound like you want parse whole `github`
it gonna take a while... 🗓

File: 3135/1/gem_class.rb [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+# :reek:InstanceVariableAssumption, :reek:TooManyInstanceVariables, :reek:TooManyStatements
+class GemData
+  attr_reader :name, :rating, :watch, :star, :fork, :issues, :contributors, :used_by
+
+  def initialize(name)
+    @name = name
+  end
+
+  def call_rubygems_api
+    uri = "https://rubygems.org/api/v1/gems/#{@name}.json"
```
Comment (longbeardmen) [inline 3135/1/gem_class.rb]: you should build url with help of gem URI instead of manual processing

File: 3135/1/gem_class.rb [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  # rubocop:disable Metrics/AbcSize, Security/Open
+  # :reek:DuplicateMethodCall
+  def github_parse
+    doc = Nokogiri::HTML(open(@github))
+    @watch = doc.css('.social-count')[0].text.gsub(/\D/, '')
+    @star = doc.css('.social-count')[1].text.gsub(/\D/, '')
+    @fork = doc.css('.social-count')[2].text.gsub(/\D/, '')
+    @issues = doc.css('.Counter')[0].text.gsub(/\D/, '')
+    @contributors = doc.css("span[class='num text-emphasized']")[3].text.gsub(/\D/, '')
+
```
Comment (longbeardmen) [inline 3135/1/gem_class.rb]: lines 32-36 do same things.
Move this to separate method

File: 3135/1/auxiliary.rb [hunk 7]
```diff
@@ -0,0 +1,27 @@
+require 'YAML'
+require 'table_print'
+
+private
+
+def load_list(options)
+  YAML.load_file(options[:file] || 'gem_list.yml')['gems']
```
Comment (longbeardmen) [inline 3135/1/auxiliary.rb]: you have no file `gem_list.yml`

Comment (IvanShamatov) [reply 3135/1/auxiliary.rb]: mha?

**Pull request merged**